gameobject.cpp

```diff
@@ -19,4 +15,66 @@ void GameObject::OnUpdate()
 {
 }
 
+void GameObject::OnCollisionStart(GameObject& /*other*/)
+{
+}
+
+void GameObject::OnCollisionStay(GameObject& /*other*/)
+{
+}
+
+void GameObject::OnCollisionStop(GameObject& /*other*/)
+{
+}
+
+int GameObject::GetId() const
+{
+    return mId;
+}
+
+ColliderBox* GameObject::GetColliderBox() const
+{
+    return mColliderBox.get();
+}
+
+glm::vec3 GameObject::GetPosition() const
+{
+    return mPosition;
+}
+
+void GameObject::SetPosition(const glm::vec3& position)
+{
+    mPosition = position;
+
+    if (mColliderBox != nullptr)
+    {
+        mColliderBox->UpdatePositionBounds(mPosition);
+    }
+}
+
+const std::string& GameObject::GetName() const
```

Comment:
Consider just returning the string by value (copying). I doubt this is a perf bottleneck and if someone holds on to this reference longer than the existence of the object you may be in pain.